File: Код спорта/2026/H/H.cpp
/* authors:
 * 1) Danila "akshin_" Axyonov;
 * 2) Alexander "ValeevAlex" Valeev;
 * 3) Nikolay Osipenko.
 */

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
typedef long long int lli;
typedef pair<lli, lli> plli;

#define num first
#define ind second

void solve();

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    solve();

    return 0;
}

void solve() {
    lli n;
    cin >> n;
    vector<plli> a(n);
    for (int i = 0; i < (int)n; ++i) {
        cin >> a[i].num;
        a[i].ind = (lli)i;
    }
    sort(a.begin(), a.end());

    lli mx = 0LL;
    for (int i = 0; i < (int)n; ++i)
        if ((lli)i < a[i].ind)
            mx = max(mx, a[i].ind - (lli)i);
    cout << mx << '\n';
}

File: HackIT/2026/regional/H/H.py
# authors:
# 1) Danila "akshin_" Axyonov;
# 2) Alexander "ValeevAlex" Valeev;
# 3) Nikita "rizhiy" Terebinov.

n = int(input())
a = {}
for _ in range(n):
    s = input()
    s1, s2 = s.split(' -> ')
    s3 = s2.split(', ')
    a[s1] = s3
q = []
used = set()
l = r = 0
q.append('305'); used.add('305'); r += 1
q.append('309'); used.add('309'); r += 1
while l < r:
    cur = q[l]; l += 1
    for nb in a.get(cur, []):
        if nb in used:
            continue
        q.append(nb); used.add(nb); r += 1
print('YES' if '000' in used else 'NO')
